virl2_client/models/cl_pyats.py

```diff
@@ -267,7 +313,27 @@ def run_config_command(
         )
 
     def cleanup(self) -> None:
-        """Clean up the pyATS connections."""
-        for pyats_device in self._connections:
-            pyats_device.destroy()
-        self._connections.clear()
+        """Clean up all pyATS connections."""
+        for pyats_device in tuple(self._connections):
+            try:
+                pyats_device.destroy()
+            finally:
+                self._connections.discard(pyats_device)
+
+    def cleanup_node_connection(self, node_label: str) -> None:
+        """
+        Clean up the pyATS connection for a specific node.
+
+        :param node_label: The label/title of the node whose connection to cleanup
+        """
+        if self._testbed is None:
+            return
+        try:
+            pyats_device: "Device" = self._testbed.devices[node_label]
+        except Exception:
```

Comment:
what other errors are possible?
```suggestion
        except KeyError:
```